core-war/src/main/webapp/attachment/jsp/displayAttachedFiles.jsp

```diff
@@ -1352,32 +1280,34 @@
     createSharingTicketPopup(sharingObject);
   }
 
-  function notifyAttachment(attachmentId) {
+  _self.notifyAttachment = function(attachmentId) {
     alertUsersAttachment(attachmentId); //dans publication.jsp
   }
 
-  var customProtocolCookieName = "Silverpeas_OnlineEditing_CustomProtocol";
-  function showInformationAboutOnlineEditingWithCustomProtocol(id, lang) {
-    var customProtocolCookieValue = $.cookie(customProtocolCookieName);
-    if (${onlineEditingWithCustomProtocolAlert} && ("IKnowIt" != customProtocolCookieValue)) {
-      $("#dialog-attachment-onlineEditing-customProtocol").data({
-        'docId': id,
-        'lang':lang}).dialog("open");
+  const customProtocolCookieName = "Silverpeas_OnlineEditing_CustomProtocol";
+  function _showInformationAboutOnlineEditingWithCustomProtocol(id, lang) {
+    const customProtocolCookieValue = $.cookie(customProtocolCookieName);
+    if (${onlineEditingWithCustomProtocolAlert} && ("IKnowIt" !== customProtocolCookieValue)) {
+      _self.ui.get$onlineEditingCustomProtocol().data({
+        'docId' : id,
+        'lang' : lang
+      }).dialog("open");
     } else {
-      openDocViaCustomProtocol(id, lang);
+      _openDocViaCustomProtocol(id, lang);
     }
   }
 
-  function getOnlineEditionLauncherURL(docId, lang) {
+  const _getOnlineEditionLauncherURL = function(docId, lang) {
     return "<%=URLUtil.getFullApplicationURL(request)%>/attachment/jsp/launch.jsp?id="+docId+"&lang="+lang;
   }
 
-  function openDocViaCustomProtocol(docId, lang) {
-    $.get(getOnlineEditionLauncherURL(docId, lang));
+  const _openDocViaCustomProtocol = function(docId, lang) {
+    $.get(_getOnlineEditionLauncherURL(docId, lang));
   }
+};
 </script>
 
-<div id="dialog-attachment-update" style="display:none">
+<div id="dialog-attachment-update" class="dialog-attachment-update${domIdSuffix}" style="display:none">
```

Comment:
When _displayAttachedFiles.jsp_ is invoked several times in a single page (one per attachment block), the identifier of the HTML elements here and below isn't anymore unique. This is against the HTML standard. If CSS is here on such identifier, it would be nice to invert the name (and hence the behavior) between CSS class and identifier.
That is to say the CSS in on the class whose name doesn't change with the attachment block it is related to, and the JS (`_afManager${domIdSuffix}`) is on the identifier whose name, by consequence, change with the attachment block (for example `id="dialog-attachment-update${domIdSuffix}" class="dialog-attachment-update"`)

Reply:
Yes, I know that.
And also that what I've done isn't enough.
But there are several CSS statements mapped to the id attribute of this JSP. And probably many more defined specifically on the client servers.
So I focused here only on the Javascript part in order to have several instances with their own scope.